modules/gltf/test/lib/extensions/EXT_mesh_features.spec.ts

```diff
@@ -1,181 +1,289 @@
+  );
+  t.end();
+});
+
+const PRIMITIVE_EXPECTED = {
+  attributes: {TEXCOORD_0: 2, POSITION: 1, _FEATURE_ID_0: 3},
+  indices: 0,
+  material: 0,
+  extensions: {
+    EXT_mesh_features: {
+      featureIds: [
         {
-          bufferView: 1,
-          byteOffset: 0,
-          componentType: 5126,
-          count: 4,
-          type: 'VEC3',
-          max: [48149.36831235699, 47655.16766258143, 53559.4425966118],
-          min: [-48149.36831235606, -47655.16766258143, -53559.4425966118]
+          featureCount: 1,
+          texture: {
+            index: 1
+          },
+          propertyTable: 0
         },
         {
-          bufferView: 2,
-          byteOffset: 0,
-          componentType: 5126,
-          count: 4,
-          type: 'VEC2'
+          featureCount: 7,
+          propertyTable: 8,
+          attribute: 0
         }
-      ],
-      extensions: {},
-      extensionsUsed: ['EXT_mesh_features'],
-      meshes: [
-        {
-          primitives: [
-            {
-              attributes: {
-                TEXCOORD_0: 2,
-                POSITION: 1
-              },
-              indices: 0,
-              material: 0,
-              extensions: {
-                EXT_mesh_features: {
-                  featureIds: [
-                    {
-                      featureCount: 1,
-                      texture: {
-                        index: 1
-                      },
-                      propertyTable: 0,
-                      data: undefined
-                    }
-                  ]
-                }
-              }
-            }
-          ]
-        }
-      ],
-      textures: [
+      ]
+    }
+  }
+};
+
+const PRIMITIVE_DECODED_EXPECTED = {
+  attributes: {TEXCOORD_0: 2, POSITION: 1, _FEATURE_ID_0: 3},
+  indices: 0,
+  material: 0,
+  extensions: {
+    EXT_mesh_features: {
+      featureIds: [
         {
-          sampler: 0,
-          source: 0
+          featureCount: 1,
+          texture: {
+            index: 1
+          },
+          propertyTable: 0,
+          data: [1, 1, 1, 1]
         },
         {
-          sampler: 1,
-          source: 1
+          featureCount: 7,
+          propertyTable: 8,
+          data: {
+            '0': 4,
+            '1': 4,
+            '2': 4,
+            '3': 3,
+            '4': 0,
+            '5': 1,
+            '6': 2
+          },
+          attribute: 0
         }
       ]
     }
-  };
+  }
+};
+
+test('gltf#EXT_mesh_features - Should encode featureIDs', (t) => {
+  const gltf = JSON.parse(JSON.stringify(GLTF_WITH_EXTENSION));
+  gltf.buffers[0].arrayBuffer = new Uint8Array(binaryBufferData).buffer;
+
+  const scenegraph = new GLTFScenegraph(gltf as unknown as {json: GLTF});
+  const primitive = gltf.json.meshes[0].primitives[0];
+  const featureIds = [4, 4, 4, 3, 0, 1, 2];
+  const typedArray = new Uint32Array(featureIds);
+  const tableIndex = 8;
+
+  createExtMeshFeatures(scenegraph, primitive, typedArray, tableIndex);
```

Comment:
The extensions should have interfaces "decode" and "encode". `createExtMeshFeature` is not necessary

Reply:
As discussed the removing this function results in the necessity to put all input data into GLTF which complicates the implementation.